azure-pipelines.yml

```diff
@@ -0,0 +1,112 @@
+trigger: none
+
+resources:
+  repositories:
+    - repository: AksEdge
+      type: github
+      endpoint: AksEdge-GithubConnection
+      name: Azure/AKS-Edge
+      ref: $(Build.SourceBranch)
+
+jobs:
+  - job: aidescript_e2e_test
+    displayName: Run E2E Aide Test Script
+    pool: 1es-aksiot-windows-x64-ltsc-2021-test-pool
+    timeoutInMinutes: 60
+    
+    steps:
+      - checkout: AksEdge
+        path: self
+        clean: true
+        fetchDepth: 1
+        
+      - powershell: |
+          $CheckInstaller = Get-WmiObject -Class Win32_Product | where Name -match "AKS Edge Essentials - (K8s|K3s)"
+          $Module = Get-Module -ListAvailable -Name AksEdge
+          if ($CheckInstaller -ne $Null -Or $Module -ne $Null)
+          {
+            Write-Error "AksEdge is already installed on this agent"
+            Exit 1
+          }
+          Write-Host "AksEdge is not installed on this agent"
+
+          $freememInMB = ((Get-CimInstance -Class Win32_OperatingSystem).FreePhysicalMemory / 1024)
+          $freememInMBRounded = [Math]::Round($freememInMB)
+          if ($freememInMbRounded -lt 4096)
+          {
+            Write-Error "The host does not have enough resources to install and run AksEdge"
+            Exit 1
+          }
+          Write-Host "The host has $freememInMBRounded free memory, AksEdge can be installed on it"
+
+          $PSConfiguration = Get-ExecutionPolicy
+          if ($PSConfiguration -ne "Bypass" -and $PSConfiguration -ne "Unrestricted")
+          {
+            Write-Error "The host current powershell configuration is $PSConfiguration, expected configuration is Bypass or Unrestricted"
+            Exit 1
+          }
+          Write-Host "The host current powershell configuration is $PSConfiguration"
+
+          $SSHCheck = (Get-WindowsCapability -Online | Where-Object Name -like 'OpenSSH.Client*').State
```

Comment:
Maybe it's worth also checking the correct SSH version - We do check that as part of our installer, but maybe we should add it here

Reply:
We could add it but not sure if it's really needed. The only reason to add it here is so we could catch the error earlier.